### PR TITLE
12 MachineData entity class

### LINKED ISSUE #12: MachineData entity class
**Description of the feature to be developed:**
Make the entity class "MachineData" and add the variable fields found in the class diagram to it.

### PR DESCRIPTION
**Pull request by Kaszz**
Closes #12 

## Description
Added the MachineData entity class with its attributes, constructor, getters, and setters. Also wrote test for the methods.

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes, if necessary.
- [x] All tests pass (existing and potential new).
- [x] Sonar check has passed
- [x] No codesmells or bugs

## Other Relevant Information
I had to use the //NOSONAR comment to ignore a code smell that didn't like having more than 7 attributes in the constructor.

### CODE REVIEW

**Review by Wylardt — APPROVED** [review]

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/entities/MachineData.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    private double humidity;
+    private Ingredients ingredients;
+    private int acceptableProducts;
+    private int defectProducts;
+    private int processed;
+    private double maintenance;
+    private LocalDateTime timestamp;
+
+
+    public MachineData(double normSpeed, int state, double temperature, double vibration, double humidity, Ingredients ingredients, int acceptableProducts, int defectProducts, int processed, double maintenance, LocalDateTime timestamp) { //NOSONAR
```
Comment (Teyson) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/entities/MachineData.java:19]: 🤮

File: BrewMesModulesParent/common/src/test/java/com/brewmes/common/entities/MachineDataTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+class MachineDataTest {
+
+    MachineData machineData;
+
+    @BeforeEach
+    void setUp() {
+        machineData = new MachineData(50.0, 2, 2.0, 3.0, 4.0,
+                new Ingredients(100.0, 200.0, 300.0, 400.0, 500.0),
+                25, 10, 35, 20.0, LocalDateTime.of(1995, 5, 2, 3, 25));
```
Comment (Teyson) [inline BrewMesModulesParent/common/src/test/java/com/brewmes/common/entities/MachineDataTest.java:18]: 🤮

File: BrewMesModulesParent/common/src/main/java/com/brewmes/common/entities/MachineData.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        this.state = state;
+        this.temperature = temperature;
+        this.vibration = vibration;
+        this.humidity = humidity;
+        this.ingredients = ingredients;
+        this.acceptableProducts = acceptableProducts;
+        this.defectProducts = defectProducts;
+        this.processed = processed;
+        this.maintenance = maintenance;
+        this.timestamp = timestamp;
```
Comment (Teyson) [inline BrewMesModulesParent/common/src/main/java/com/brewmes/common/entities/MachineData.java:30]: 🤮

**Review by Teyson — APPROVED** [review]

**Review by dottore1 — APPROVED** [review]

**Review by Lawst96 — APPROVED** [review]
Thanks, I hate it.

**Review by Kechr — APPROVED** [review]

**Pull request merged**